api/src/services/authentication-service.ts

```diff
@@ -215,28 +215,28 @@ export class AuthenticationService {
 		return trustedRoots.map((root) => root.userId);
 	};
 
-	getChallenge = async (userId: string) => {
+	getNonce = async (userId: string) => {
 		const user = await this.userService.getUser(userId);
 		if (!user) {
 			throw new Error(`no user with id: ${userId} found!`);
 		}
 
-		const challenge = createChallenge();
-		await AuthDb.upsertChallenge({ userId: user.userId, challenge });
-		return challenge;
+		const nonce = createNonce();
+		await AuthDb.upsertNonce(user.userId, nonce);
+		return nonce;
 	};
 
-	authenticate = async (signedChallenge: string, userId: string) => {
+	authenticate = async (signedNonce: string, userId: string) => {
```

Comment:
It would probably make sense to define a type with Typebox where you can specify expected length of the `signedNonce`